src/sfizz/Voice.cpp

```diff
@@ -869,16 +862,14 @@ void sfz::Voice::setupOscillatorUnison()
 
 void sfz::Voice::updateChannelPowers(AudioSpan<float> buffer)
 {
-    assert(smoothedChannelEnvelopes.size() == channelEnvelopeFilters.size());
-    assert(buffer.getNumChannels() <= channelEnvelopeFilters.size());
     if (buffer.getNumFrames() == 0)
         return;
 
-    for (unsigned i = 0; i < smoothedChannelEnvelopes.size(); ++i) {
+    const float factor = buffer.getNumFrames() / samplesPerBlock * trackingFactor;
+    for (unsigned i = 0; i < meanChannelPowers.size(); ++i) {
         const auto input = buffer.getConstSpan(i);
-        for (unsigned s = 0; s < buffer.getNumFrames(); ++s)
-            smoothedChannelEnvelopes[i] =
-                channelEnvelopeFilters[i].tickLowpass(std::abs(input[s]));
+        const float meanPower = sfz::meanSquared(input);
+        meanChannelPowers[i] = meanChannelPowers[i] * (1 - factor) + meanPower * factor;
```

Comment:
The analysis signal is not windowed at the input.
In former experience, RMS value of raw input can be very imprecise.

Reply:
Windowed as in https://en.wikipedia.org/wiki/Hann_function ?

Reply:
We need *some* indication or the power btw, it does not need to be too precise.

Reply:
Yes, I would once find the measurement to be very noisy, (this was another project out of sfizz)
and the curve next to unusable.
Imo, at the first look, I'm not feeling convinced by this method.
It's tied to the samples per block. If the signal is too low frequency, and the block size is short, the method seems likely to fail.

Reply:
It's supposed to be somewhat independent on the samples per block, the factor changes depending on the expected block size and the actual one. I will double check since I did this at 2 in the morning.

I don't see how windowing would help though, it will bias the average? The noisiness can be smoothed away, the issue being that attacks and releases are not tracked quickly enough. Apparently people also take the max of 2 parallel filters for such cases, with different responses, to track attacks fast and releases less so.